artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java

```diff
@@ -79,4 +85,81 @@ public static List<String> prepareMavenProperties() {
         return extraProperties;
     }
 
+    /**
+     * Check for maven.repo.local property to propagated it. If the property is not set the default M2 home
+     * @return String path to local repository
+     */
+    public static String getLocalRepository() {
+        String localRepo = Objects.requireNonNullElse(System.getProperty("maven.repo.local"), "");
+        if (localRepo.isEmpty()) {
+            localRepo = "~/.m2/repository/";
```

Comment:
`~` is bash/shell thingy 

`System.getProperty("user.home")` should be more safe + you should consider this to be also Windows friendly

Reply:
Good catch! Updated it to be more safe on multiplatform.